app/views/samples/_upload_results_js.haml

```diff
@@ -0,0 +1,125 @@
+        reader.onload = (function(f) {
+            return async(load_event) => {
+                // Parse the input file
+                var text =  load_event.target.result
+                var lines = text.split(/\r\n|\r|\n/); // tolerate both Windows and Unix linebreaks
+                var samples_param = [] // params for GET call
+                var samples = [] // samples uuids
+
+                lines.forEach(function(line) { 
+                    if (line != ''){
+                        samples_param.push( 'uuids[]=' + line.split(',')[0] );
+                        samples.push( line.split(',')[0] );
+                    } 
+                });
+                
+                // Retrieve found uuids
+                var context = getUrlParameter('context');
+                var url = '/samples/existing_uuids?context='+context+"&"+samples_param.join("&")
+                var found = await fetch(url)
+                    .then((response) => response.json())
+                    .then((r) => {
+                        return r.message;
+                })
+                // Compute not found ones
+                var not_found = samples.filter(function(x) {
+                    return found.indexOf(x) < 0;
+                });
+                
+                // Create row from template with filename and upload info
+                var template = document.getElementById(`csvFileRow`).content;
+                var fragment = template.cloneNode(true);
+                fragment.querySelector(`.uploaded-samples-count`).textContent = `${samples.length} Samples ` ;
+                if (found.length != samples.length){
+                    fragment.querySelector(`.uploaded-samples-count`).innerHTML += `(<span class="dashed-underline">${(not_found.length)} sample UUID${(not_found.length>1?'s':'')}</span> not found) `
+                    fragment.querySelector(`.upload-icon`).classList.add(`icon-alert`,`icon-red`)
+                    fragment.querySelector(`.samples-row-action`).classList.add(`ttip`,`input-required`)
+                }
+                else {
+                    fragment.querySelector(`.upload-icon`).classList.add(`icon-checkmark`)
+                }
+                fragment.querySelector(`.file-name`).textContent = `${f.name}`;
+                // if samples were not found add mouse behaviour to display tooltip
+                if (found.length != samples.length) {
+                    var showToolTip = function(evt) {
+                            var tooltip = evt.currentTarget.tooltip
+                            if (tooltip.classList.contains("hidden")) {
+                                tooltip.classList.remove("hidden")
+                                setTimeout(() => tooltip.classList.add("hidden"), 5000)
+                            }
+                            else {
+                                tooltip.classList.toggle("hidden")
+                            }
+                        }
```

Comment:
You could extract the `showTooltip` function out of `handleFileSelect`. It looks fully generic to me.